pandas/tests/indexes/datetimes/test_timezones.py

```diff
@@ -442,16 +442,20 @@ def test_dti_tz_localize_utc_conversion(self, tz):
     ])
     def test_dti_tz_localize_roundtrip(self, tz_aware_fixture, idx):
         tz = tz_aware_fixture
-        localized = idx.tz_localize(tz)
-        expected = date_range(start=idx[0], end=idx[-1], freq=idx.freq,
-                              tz=tz)
-        tm.assert_index_equal(localized, expected)
+        localized = idx.tz_localize(tz, ambiguous='NaT')
```

Comment:
Instead of this, keep the old test and just parametrize over ambiguous=`True`/`False`

Reply:
```
@pytest.mark.paramertize('ambiguous', [True, False])
def test_dti_tz_localize_roundtrip(self, tz_aware_fixture, idx, ambiguous):
        tz = tz_aware_fixture
        localized = idx.tz_localize(tz, ambiguous=ambiguous)
        expected = date_range(start=idx[0], end=idx[-1], freq=idx.freq).tz_localize(tz, ambiguous=ambiguous)
        tm.assert_index_equal(localized, expected)
```

Reply:
I don’t see how this assert can ever fail...

Reply:
Looks like the main point of this test is to test the roundripping not necessarily that the construction of the indexes break. So here we are just shifting ambiguous times forward or backwards to ensure this.

Reply:
The roundtripping is tested by the other assert at the end of the testcase, isnt it? Stripping away the first assert, and the handling of the special cases for a moment, the TC would reduce to this:

```
localized = idx.tz_localize(tz)
reset = idx.tz_localize(None)
tm.assert_index_equal(reset, idx)
```
The first assert tries to check `localized` directly, and feels like it should be covered by the other`test_dti_tz_localize_...` testcases. Its hard to construct `expected` from `idx` when there's a DST transition, without just duplicating `localized`.